src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -481,79 +494,90 @@ void DnsMetricsBucket::process_dnstap(bool deep, const dnstap::Dnstap &payload)
     case dnstap::Message_Type_AUTH_QUERY:
     case dnstap::Message_Type_RESOLVER_QUERY:
         side = QR::query;
-        ++_counters.queries;
         break;
     }
 
-    if (payload.message().has_query_port()) {
-        _dns_topUDPPort.update(payload.message().query_port());
-    }
-
     if (payload.message().has_query_zone()) {
         // TODO decode wire name, use in top_qname
     }
 
     if (!deep || (!payload.message().has_query_message() && !payload.message().has_response_message())) {
+        process_dns_layer(l3, l4, side, 0);
```

Comment:
@weyrick  actually it was already right the logic here. don't need the extra else that i've added